lib/SIL/IR/SILProfiler.cpp

```diff
@@ -178,11 +178,12 @@ namespace {
 ///
 /// Apply \p Func if the function can be visited.
 template <typename F>
-bool visitFunctionDecl(ASTWalker &Walker, AbstractFunctionDecl *AFD, F Func) {
+ASTWalker::PreWalkAction
+visitFunctionDecl(ASTWalker &Walker, AbstractFunctionDecl *AFD, F Func) {
   bool continueWalk = Walker.Parent.isNull();
   if (continueWalk)
     Func();
-  return continueWalk;
+  return ASTWalker::Action::VisitChildrenIf(continueWalk);
```

Comment:
Yeah good point, this isn't a `Stop` case tho, this helper is to handle e.g nested functions, to ensure we only visit their bodies when walking their specific SIL function instead of the parent function